src/pages/interviews.tsx

```diff
@@ -118,6 +118,24 @@ function Applicants({ type, applicants, interviews, refetchInterviews,
   refetchInterviews: () => void,
   refetchApplicants: () => void,
 }) {
+  // A map from applicants' user ids to sources.
+  const [sources, setSources] = useState<{ [id: string]: string }>({});
+  const updateSource = (id: string) => (source: string) => {
+    setSources(prev => {
+      if (prev[id] === source) return prev; 
+      return { ...prev, [id]: source };
+    });
+  };
+  const sortedApplicants = useMemo(() => {
+    return applicants.sort((a1, a2) => {
+      const comp = toPinyin(sources[a1.id] || '')
+        .localeCompare(toPinyin(sources[a2.id] || ''));
```

Comment:
create a function `compareChinese` in strings.ts. Add comments in this new function why calling toPinyin is needed. Call that function from here.